ttnn/cpp/ttnn/operations/experimental/auto_format/auto_format.cpp

```diff
@@ -53,19 +53,29 @@ Tensor AutoFormat::move_tensor_to_device_and_pad(
         (device_shape[-2] % TILE_HEIGHT != 0 ? (device_shape[-2] / TILE_HEIGHT + 1) * TILE_HEIGHT : device_shape[-2]),
         (device_shape[-1] % TILE_WIDTH != 0 ? (device_shape[-1] / TILE_WIDTH + 1) * TILE_WIDTH : device_shape[-1])};
     const auto new_shape = tt::tt_metal::LegacyShape(new_intended_shape, new_device_shape);
-    return AutoFormat::format_input_tensor(input, device, new_shape, 0.0, target_layout, target_mem_config);
+
+    std::variant<int, float> pad_value;
+    if(input.get_dtype() == ttnn::DataType::BFLOAT16) {
```

Comment:
why does it seem like this is copy pasted everywhere

Reply:
can this just be some standard getter function?